src/core/Threads/ThreadPool.cpp

```diff
@@ -1,41 +1,45 @@
 #include "ThreadPool.hpp"
 
-#include "ThreadWorker.hpp"
-#include "ConsoleLog.hpp"
-
 ThreadPool::ThreadPool(unsigned int threadsNumber)
-  : m_jobsPending(0)
-  , m_threads(std::vector<std::thread>(threadsNumber))
-  , m_shutdown(false)
 {
-  for (auto& thread : m_threads)
-  {
-    m_threads.emplace_back(std::thread(ThreadWorker(this)));
-  }
+  m_threads.reserve(threadsNumber);
+  m_tasks.reserve(threadsNumber);
 }
 
 ThreadPool::~ThreadPool()
 {
   Shutdown();
 }
 
-void ThreadPool::Shutdown()
+void ThreadPool::Submit(std::unique_ptr<ThreadTask> &&task)
 {
-  m_shutdown = true;
+  std::unique_ptr<sf::Thread> thread(new sf::Thread(&ThreadTask::Run, task.get()));
```

Comment:
It's a terrible thing. All the thread should be created at the very begging of the program run but I didn't find the way to do it with SFML threads.